tokio-console/src/warnings.rs

```diff
@@ -150,3 +154,56 @@ impl Warn<Task> for LostWaker {
         "This task has lost its waker, and will never be woken again.".into()
     }
 }
+
+/// Warning for if a task has never yielded
+#[derive(Clone, Debug)]
+pub(crate) struct NeverYielded {
+    min_duration: Duration,
+    description: String,
+}
+
+impl NeverYielded {
+    pub(crate) const DEFAULT_DURATION: Duration = Duration::from_millis(10);
+    pub(crate) fn new(min_duration: Duration) -> Self {
+        Self {
+            min_duration,
+            description: format!(
+                "tasks have never yielded (threshold {}ms)",
+                min_duration.as_millis()
+            ),
+        }
+    }
+}
+
+impl Default for NeverYielded {
+    fn default() -> Self {
+        Self::new(Self::DEFAULT_DURATION)
+    }
+}
+
+impl Warn<Task> for NeverYielded {
+    fn summary(&self) -> &str {
+        self.description.as_str()
+    }
+
+    fn check(&self, task: &Task) -> bool {
+        // Don't fire warning for tasks that are waiting to run
+        if task.state() != TaskState::Running {
+            return false;
+        }
+
+        if task.yielded() {
```

Comment:
I would inline the logic from `task.yielded()` here and then we don't need that method at all.

```suggestion
        if task.total_polls() > 1 {
```